pkg/logs/tailers/file/tailer.go

```diff
@@ -383,6 +383,8 @@ func (t *Tailer) forwardMessages() {
 		}
 
 		msg := message.NewMessage(output.GetContent(), origin, output.Status, output.IngestionTimestamp)
+		// Preserve ParsingExtra information from decoder output (including IsTruncated flag)
+		msg.ParsingExtra = output.ParsingExtra
```

Comment:
Same here, I'm wondering why we need to create a new message and can't modify/reuse the existing one.